xscen/finalize.py

```diff
@@ -45,3 +50,125 @@ def change_units(ds: xr.Dataset, variables_and_units: dict):
+     - remove everything but a list of attributes
+     - add attributes
+     - change the prefix of the catalog attrs
+
+     in that order.
+
+    Parameters
+    ----------
+    ds: xr.Dataset
+        Input dataset to clean up
+    var_and_convert_units: dict
+        Dictionary of variable to convert. eg. {'tasmax': 'degC', 'pr': 'mm d-1'}
+    maybe_unstack_dict: dict
+        Dictionnary to pass to xscen.common.maybe_unstack fonction.
+        The format should be: {'coords': path_to_coord_file, 'rechunk': {'time': -1 }, 'stack_drop_nans': True}.
+    add_feb_29: bool
+        Wheter to add back a february 29th (that was removed in a noleap calendar) by linear interpolation
+    attrs_to_remove: dict
+        Dictionary where the keys are the variables and the values are a list of the attrs that should be removed.
+        For global attrs, use the key 'global'.
+        The element of the list can be exact matches for the attributes name
+        or use the same substring matching rules as intake_esm:
+        - ending with a '*' means checks if the substring is contained in the string
+        - starting with a '^' means check if the string starts with the substring.
+        eg. {'global': ['unnecessary note', 'cell*'], 'tasmax': 'old_name'}
+    remove_all_attrs_except: dict
+        Dictionary where the keys are the variables and the values are a list of the attrs that should NOT be removed,
+        all other attributes will be deleted. If None (default), nothing will be deleted.
+        For global attrs, use the key 'global'.
+        The element of the list can be exact matches for the attributes name
+        or use the same substring matching rules as intake_esm:
+        - ending with a '*' means checks if the substring is contained in the string
+        - starting with a '^' means check if the string starts with the substring.
+        eg. {'global': ['necessary note', '^cat/'], 'tasmax': 'new_name'}
+    add_attrs: dict
+        Dictionary where the keys are the variables and the values are a another dictionary of attributes.
+        For global attrs, use the key 'global'.
+        eg. {'global': {'title': 'amazing new dataset'}, 'tasmax': {'note': 'important info about tasmax'}}
+    change_attr_prefix: str
+        Replace "cat/" in the catalogue global attrs by this new string
+
+    Returns
+    -------
+    ds: xr.Dataset
+        Cleaned up dataset
+    """
+    if var_and_convert_units:
+        logger.info(f"Converting units: {var_and_convert_units}")
+        ds = change_units(ds=ds, variables_and_units=var_and_convert_units)
+
+    # unstack nans
+    if maybe_unstack_dict:
+        ds = maybe_unstack(ds, **maybe_unstack_dict)
+
+    # put back feb 29th
+    if add_feb_29:
+        logging.info("Adding back february 29th by linear interpolation")
+        with_missing = convert_calendar(ds, "standard", missing=np.NaN)
+        ds = with_missing.interpolate_na("time", method="linear")
```

Comment:
Is it always adequate? For precip, we used to simply put 0 mm.

Reply:
Mais pourquoi 0 mm ? Il me semble que si les jours avant et après sont mouillés, il y des bonnes chances que le jour du milieu soit mouillé aussi, non ?

Reply:
Par exemple, ouragan Irene le 28 août 2011: On passe de 0 --> 134 --> 4 mm sur trois jours. Ce n'est pas linéaire.
https://climat.meteo.gc.ca/climate_data/daily_data_f.html?StationID=27442&timeframe=2&StartYear=1840&EndYear=2022&Day=14&Year=2011&Month=8#

Même si on regarde plutôt février, ça me semble rarement linéaire.

Reply:
hum ouin ok j'avoue que ce n'est pas super. Je ne sais pas trop quoi faire.

Reply:
Souvent, rester dans le calendrier non-standard est une solution adéquate, non?
Mais je pense que je vote pour soit NaN soit 0 ici (pour les variables du genre precip).